python/tank/authentication/user.py

```diff
@@ -118,6 +126,93 @@ def impl(self):
         return self._impl
 
 
+class ShotgunSamlUser(ShotgunUser):
+    """
+    This specialized shotgun user is needed when SSO is used, as it provides
+    mechanisms
```

Comment:
Comment not completed. It would be nice if the class explained how to use the claims methods as well.